networklibrary/src/main/java/com/eis/smsnetwork/SMSNetDictionary.java

```diff
@@ -1,39 +1,46 @@
 package com.eis.smsnetwork;
 
+import androidx.annotation.NonNull;
 
 import com.eis.communication.network.NetDictionary;
+import com.eis.smsnetwork.broadcast.BroadcastReceiver;
 
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
 
 /**
  * Concrete implementation of a NetDictionary
  *
  * @author Marco Cognolato
+ * @author Giovanni Velludo
  */
 public class SMSNetDictionary implements NetDictionary<String, String> {
 
-    Map<String, String> dict = new HashMap<>();
+    private final Map<String, String> dict = new HashMap<>();
 
     /**
      * Adds a resource to the network dictionary
      *
      * @param key      The key which defines the resource
      * @param resource The resource to add
-     * @throws IllegalArgumentException If the key is invalid.
-     *                                  A key is said to be valid only if it's composed of a single word
+     * @throws IllegalArgumentException if one of the arguments contains a backslash as its last
+     *                                  character.
      */
     public void addResource(String key, String resource) {
         checkKeyValidity(key);
+        checkKeyValidity(resource);
         dict.put(key, resource);
     }
 
     /**
      * Removes a resource from the dictionary
      *
      * @param key The key which defines the resource
-     * @throws IllegalArgumentException If the key is invalid.
-     *                                  A key is said to be valid only if it's composed of a single word
+     * @throws IllegalArgumentException if the argument contains a backslash as its last
+     *                                  character.
      */
     public void removeResource(String key) {
         checkKeyValidity(key);
```

Comment:
i think that in removing resources there's not benefits in checking the key validity
if is not valid then the default value for not found is returned cause the resource cannot be in the dictionary